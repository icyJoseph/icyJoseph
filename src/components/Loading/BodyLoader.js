import React from "react";
import { Segment, Loader as Spinner } from "semantic-ui-react";
import { Message } from "semantic-ui-react";
import Broken from "../../logos/Broken";

export const BodyLoader = ({ pastDelay, error }) => {
  // Handle the error state
  if (error) {
    return (
      <Segment inverted style={style}>
        <Broken />
        <Message
          error
          style={{
            marginTop: "-100px",
            borderRadius: 0,
            boxShadow: "none"
          }}
          color="black"
        >
          <Message.Content>
            <Message.Header>
              There was a problem loading this page.
            </Message.Header>
            Please reload the page.
          </Message.Content>
        </Message>
      </Segment>
    );
  } else if (pastDelay) {
    // Handle the loading state
    return (
      <Segment inverted style={style}>
        <Spinner active indeterminate size="massive" />
      </Segment>
    );
  } else {
    return null;
  }
};

const style = {
  margin: 0,
  position: "absolute",
  top: 0,
  bottom: 0,
  borderRadius: 0,
  width: "100%",
  flexL: 1,
  display: "flex",
  justifyContent: "center",
  flexDirection: "column",
  alignItems: "center"
};

export default BodyLoader;
