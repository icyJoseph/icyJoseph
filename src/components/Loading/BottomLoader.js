import React from "react";
import { Message, Icon } from "semantic-ui-react";

export const BottomLoader = ({ pastDelay, error }) => {
  // Handle the error state
  if (error) {
    return (
      <Message
        error
        style={{
          position: "absolute",
          bottom: 0,
          margin: 0,
          borderRadius: 0,
          boxShadow: "none",
          zIndex: 2
        }}
        color="black"
        icon
      >
        <Icon name="circle notched" color="grey" />
        <Message.Content>
          <Message.Header>
            There was a problem loading this component.
          </Message.Header>
          Please reload the page.
        </Message.Content>
      </Message>
    );
  } else if (pastDelay) {
    // Handle the loading state
    return (
      <Message
        style={{
          position: "absolute",
          margin: 0,
          borderRadius: 0,
          bottom: 0,
          zIndex: 2
        }}
        icon
        color="black"
      >
        <Icon name="circle notched" loading color="grey" />
        <Message.Content>
          <Message.Header>Just one second...</Message.Header>
          Fetching this component for you.
        </Message.Content>
      </Message>
    );
  } else {
    return null;
  }
};

export default BottomLoader;
