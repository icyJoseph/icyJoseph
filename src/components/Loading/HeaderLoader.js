import React from "react";
import { Message, Icon } from "semantic-ui-react";

export const HeaderLoader = ({ pastDelay, error }) => {
  // Handle the error state
  if (error) {
    return (
      <Message error style={{ margin: 0 }} color="black" icon>
        <Icon name="circle notched" color="grey" />
        <Message.Content>
          <Message.Header>
            There was a problem loading this component.
          </Message.Header>
          Please reload the page.
        </Message.Content>
      </Message>
    );
  } else if (pastDelay) {
    // Handle the loading state
    return (
      <Message style={{ margin: 0, borderRadius: 0 }} icon color="black">
        <Icon name="circle notched" loading color="grey" />
        <Message.Content>
          <Message.Header>Just one second...</Message.Header>
          Fetching this component for you.
        </Message.Content>
      </Message>
    );
  } else {
    return null;
  }
};

export default HeaderLoader;
