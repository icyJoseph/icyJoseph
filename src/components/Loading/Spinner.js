import React from "react";

export const Spinner = ({ pastDelay, error }) => {
  return pastDelay ? <Loader active inline="centered" /> : null;
};

export default Spinner;
