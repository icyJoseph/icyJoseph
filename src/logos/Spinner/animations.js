import { css, keyframes } from "styled-components";

export const rotate = keyframes`
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(270deg);
    }
  `;

export const SmallDash = keyframes`
    0% {
      stroke-dashoffset: 100;
    }
    50% {
      stroke-dashoffset: 50;
      transform: rotate(135deg);
    }
    100% {
      stroke-dashoffset: 100;
      transform: rotate(450deg);
    }
  `;

export const MediumDash = keyframes` 
    0% {
      stroke-dashoffset: 150;
    }
    50% {
      stroke-dashoffset: 50;
      transform: rotate(135deg);
    }
    100% {
      stroke-dashoffset: 150;
      transform: rotate(450deg);
    }

`;

export const LargeDash = keyframes` 
    0% {
      stroke-dashoffset: 200;
    }
    50% {
      stroke-dashoffset: 50;
      transform: rotate(135deg);
    }
    100% {
      stroke-dashoffset: 200;
      transform: rotate(450deg);
    }
  `;

export const SmallSpinnerPath = css`
  stroke-dasharray: 100;
  stroke-dashoffset: 0;
  transform-origin: center;
  animation: ${SmallDash} 1.3s ease-in-out infinite;
`;

export const MediumSpinnerPath = css`
  stroke-dasharray: 150;
  stroke-dashoffset: 0;
  transform-origin: center;
  animation: ${MediumDash} 1.3s ease-in-out infinite;
`;

export const LargeSpinnerPath = css`
  stroke-dasharray: 200;
  stroke-dashoffset: 0;
  transform-origin: center;
  animation: ${LargeDash} 1.3s ease-in-out infinite;
`;
