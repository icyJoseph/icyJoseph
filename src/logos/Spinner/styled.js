import styled from "styled-components";

import {
  rotate,
  SmallSpinnerPath,
  MediumSpinnerPath,
  LargeSpinnerPath
} from "./animations";

const animations = {
  small: SmallSpinnerPath,
  medium: MediumSpinnerPath,
  large: LargeSpinnerPath
};

export const SpinnerContainer = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const SvgContainer = styled.svg`
  animation: ${rotate} 1.3s linear infinite;
`;

export const SvgCircle = styled.circle`
  ${props => animations[props.size]};
`;
