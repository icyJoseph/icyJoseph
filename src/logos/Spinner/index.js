import React from "react";
import { SpinnerContainer, SvgContainer, SvgCircle } from "./styled";

const SPINNER_SIZES = {
  small: 30,
  medium: 50,
  large: 70
};

const STROKE_WIDTHS = {
  small: 4,
  medium: 5,
  large: 6
};

// Heavily inspired by https://codepen.io/mrrocks/pen/EiplA
export default function Spinner({ size = "medium" }) {
  const baseSize = SPINNER_SIZES[size];
  const pathSize = baseSize / 2;
  const strokeWidth = STROKE_WIDTHS[size];
  const pathRadius = `${baseSize / 2 - strokeWidth}px`;

  return (
    <SpinnerContainer>
      <SvgContainer
        width={baseSize}
        height={baseSize}
        viewBox={`0 0 ${baseSize} ${baseSize}`}
      >
        <SvgCircle
          size={size}
          fill="none"
          stroke="white"
          strokeWidth={strokeWidth}
          strokeLinecap="flat"
          cx={pathSize}
          cy={pathSize}
          r={pathRadius}
        />
      </SvgContainer>
    </SpinnerContainer>
  );
}
