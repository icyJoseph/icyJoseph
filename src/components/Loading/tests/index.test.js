import React from "react";
import { shallow } from "enzyme";
import { HeaderLoader } from "../HeaderLoader";
import { BodyLoader } from "../BodyLoader";
import { BottomLoader } from "../BottomLoader";
import { Spinner } from "../Spinner";

describe("HeaderLoader", () => {
  it("pastDelay", () => {
    const props = {
      pastDelay: true,
      error: null
    };
    const wrapper = HeaderLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: true,
      error: "404"
    };
    const wrapper = HeaderLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: false,
      error: null
    };
    const wrapper = HeaderLoader(props);
    expect(wrapper).toBe(null);
  });
});

describe("BodyLoader", () => {
  it("pastDelay", () => {
    const props = {
      pastDelay: true,
      error: null
    };
    const wrapper = BodyLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: true,
      error: "404"
    };
    const wrapper = BodyLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: false,
      error: null
    };
    const wrapper = BodyLoader(props);
    expect(wrapper).toBe(null);
  });
});

describe("BottomLoader", () => {
  it("pastDelay", () => {
    const props = {
      pastDelay: true,
      error: null
    };
    const wrapper = BottomLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: true,
      error: "404"
    };
    const wrapper = BottomLoader(props);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: false,
      error: null
    };
    const wrapper = BottomLoader(props);
    expect(wrapper).toBe(null);
  });
});

describe("Spinner", () => {
  it("pastDelay", () => {
    const props = {
      pastDelay: true,
      error: null
    };
    const wrapper = shallow(<Spinner {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: true,
      error: "404"
    };
    const wrapper = shallow(<Spinner {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("error", () => {
    const props = {
      pastDelay: false,
      error: null
    };
    const wrapper = shallow(<Spinner {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
