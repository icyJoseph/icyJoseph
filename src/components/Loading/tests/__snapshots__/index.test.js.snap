// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyLoader error 1`] = `
<Segment
  inverted={true}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 0,
      "bottom": 0,
      "display": "flex",
      "flexDirection": "column",
      "flexL": 1,
      "justifyContent": "center",
      "margin": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <Broken />
  <Message
    color="black"
    error={true}
    style={
      Object {
        "borderRadius": 0,
        "boxShadow": "none",
        "marginTop": "-100px",
      }
    }
  >
    <MessageContent>
      <MessageHeader>
        There was a problem loading this page.
      </MessageHeader>
      Please reload the page.
    </MessageContent>
  </Message>
</Segment>
`;

exports[`BodyLoader pastDelay 1`] = `
<Segment
  inverted={true}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 0,
      "bottom": 0,
      "display": "flex",
      "flexDirection": "column",
      "flexL": 1,
      "justifyContent": "center",
      "margin": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <Loader
    active={true}
    indeterminate={true}
    size="massive"
  />
</Segment>
`;

exports[`BottomLoader error 1`] = `
<Message
  color="black"
  error={true}
  icon={true}
  style={
    Object {
      "borderRadius": 0,
      "bottom": 0,
      "boxShadow": "none",
      "margin": 0,
      "position": "absolute",
      "zIndex": 2,
    }
  }
>
  <Icon
    as="i"
    color="grey"
    name="circle notched"
  />
  <MessageContent>
    <MessageHeader>
      There was a problem loading this component.
    </MessageHeader>
    Please reload the page.
  </MessageContent>
</Message>
`;

exports[`BottomLoader pastDelay 1`] = `
<Message
  color="black"
  icon={true}
  style={
    Object {
      "borderRadius": 0,
      "bottom": 0,
      "margin": 0,
      "position": "absolute",
      "zIndex": 2,
    }
  }
>
  <Icon
    as="i"
    color="grey"
    loading={true}
    name="circle notched"
  />
  <MessageContent>
    <MessageHeader>
      Just one second...
    </MessageHeader>
    Fetching this component for you.
  </MessageContent>
</Message>
`;

exports[`HeaderLoader error 1`] = `
<Message
  color="black"
  error={true}
  icon={true}
  style={
    Object {
      "borderRadius": 0,
      "boxShadow": "none",
      "margin": 0,
      "position": "absolute",
      "top": 0,
      "zIndex": 2,
    }
  }
>
  <Icon
    as="i"
    color="grey"
    name="circle notched"
  />
  <MessageContent>
    <MessageHeader>
      There was a problem loading this component.
    </MessageHeader>
    Please reload the page.
  </MessageContent>
</Message>
`;

exports[`HeaderLoader pastDelay 1`] = `
<Message
  color="black"
  icon={true}
  style={
    Object {
      "borderRadius": 0,
      "margin": 0,
      "position": "absolute",
      "top": 0,
      "zIndex": 2,
    }
  }
>
  <Icon
    as="i"
    color="grey"
    loading={true}
    name="circle notched"
  />
  <MessageContent>
    <MessageHeader>
      Just one second...
    </MessageHeader>
    Fetching this component for you.
  </MessageContent>
</Message>
`;

exports[`Spinner error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    error="404"
    pastDelay={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "inline": "centered",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "inline": "centered",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Spinner error 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    error={null}
    pastDelay={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Spinner pastDelay 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    error={null}
    pastDelay={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "inline": "centered",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "inline": "centered",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
